catalog/app/containers/Bucket/PackageDialog/PackageDialog.js

```diff
@@ -268,6 +316,51 @@ export function MetaInput({ schemaError, input, meta, schema }) {
     changeText,
   ])
 
+  const { push: notify } = Notifications.use()
+  const [locked, setLocked] = React.useState(false)
+  // used to force json editor re-initialization
+  const [jsonEditorKey, setJsonEditorKey] = React.useState(1)
+
+  const onDrop = React.useCallback(
+    ([file]) => {
+      if (file.size > MAX_META_FILE_SIZE) {
+        notify(
+          <>
+            File too large ({readableBytes(file.size)}), must be under{' '}
+            {readableBytes(MAX_META_FILE_SIZE)}.
+          </>,
+        )
+        return
+      }
+      setLocked(true)
+      readFile(file)
+        .then((contents) => {
+          try {
+            JSON.parse(contents)
+          } catch (e) {
+            notify('The file does not contain valid JSON')
+          }
+          changeText(contents)
+          // force json editor to re-initialize
+          setJsonEditorKey(R.inc)
```

Comment:
Consider making a hook for this pattern. 

`const [cacheKey, updateCacheKey] = useReactCache()`

Reply:
i'm not sure there's a lot to gain from `incKey()` vs `setKey(R.inc)`

Reply:
* to not spend time on what is already implemented
* to have business logic and internal technical stuff (dealing with React keys) divided
* to be ready for changes in one place when the environment will change (major React update, migrating to different UI library)